platform/ui/src/components/InputRange/InputRange.tsx

```diff
@@ -38,72 +42,76 @@ const InputRange: React.FC<{
+      onChange(val);
     },
-    [onChange, setRangeValue]
+    [onChange]
   );
 
   const rangeValuePercentage = ((rangeValue - minValue) / (maxValue - minValue)) * 100;
 
-  const rangeValueForStr = step >= 1 ? rangeValue.toFixed(0) : rangeValue.toFixed(1);
+  const LabelOrEditableNumber = allowNumberEdit ? (
+    <InputNumber
+      minValue={minValue}
+      maxValue={maxValue}
+      value={rangeValue}
+      onChange={val => {
+        setRangeValue(val);
+        onChange(val);
+      }}
+      step={step}
+      showAdjustmentArrows={showAdjustmentArrows}
+    />
+  ) : (
+    <span className={classNames(labelClassName ?? 'text-white')}>
+      {value}
+      {unit}
+    </span>
+  );
 
   return (
     <div
-      className={`flex cursor-pointer items-center space-x-1 ${
-        containerClassName ? containerClassName : ''
-      }`}
+      className={`flex cursor-pointer items-center ${containerClassName ?? ''}`}
+      onClick={e => {
+        e.stopPropagation();
+        e.preventDefault();
+      }}
     >
-      {showLabel && labelPosition === 'left' && (
-        <Typography
-          variant={labelVariant ?? 'subtitle'}
-          component="p"
-          className={classNames('w-8', labelClassName ?? 'text-white')}
-        >
-          {rangeValueForStr}
-          {unit}
-        </Typography>
-      )}
-      <input
-        type="range"
-        min={minValue}
-        max={maxValue}
-        value={rangeValue}
-        className={`input-range-thumb-design h-[3px] appearance-none rounded-lg ${
-          inputClassName ? inputClassName : ''
-        }`}
-        style={{
-          background:
-            trackColor ||
-            `linear-gradient(to right, #5acce6 0%, #5acce6 ${rangeValuePercentage - 10}%, #3a3f99 ${
-              rangeValuePercentage + 10
-            }%, #3a3f99 100%)`,
-        }}
-        onChange={handleChange}
-        id="myRange"
-        step={step}
-      />
-      {showLabel && (!labelPosition || labelPosition === 'right') && (
-        <Typography
-          variant={labelVariant ?? 'subtitle'}
-          component="p"
-          className={classNames('w-8', labelClassName ?? 'text-white')}
-        >
-          {rangeValueForStr}
-          {unit}
-        </Typography>
-      )}
+      <div className="relative flex w-full items-center">
+        {showLabel && labelPosition === 'left' && LabelOrEditableNumber}
+        <div className="range-track"></div>
+        <input
+          type="range"
+          min={minValue}
+          max={maxValue}
+          value={rangeValue}
+          className={`h-[3px] appearance-none rounded-md ${inputClassName ?? ''}`}
+          style={{
+            background: `linear-gradient(to right, #5acce6 0%, #5acce6 ${rangeValuePercentage}%, #3a3f99 ${rangeValuePercentage}%, #3a3f99 100%)`,
+          }}
+          onChange={handleChange}
+          id="myRange"
+          step={step}
+        />
+        <div style={{ width: labelWidth }}>
```

Comment:
If `showLabel` is `false`, the `div` will be empty so do we still want to have this empty `div` occupy "width" space?